### PR TITLE
extend min_coin to rpc calls & cli for coin selection 

### PR DESCRIPTION
**Pull request by jack60612**
tests were added, and min_coin_amount is being added to rpc's and cli

### CODE REVIEW

**Comment by jack60612** [comment]
Manual cli tests have been successfully completed!

**Comment by jack60612** [comment]
Cat wallet support was added

File: chia/rpc/wallet_rpc_api.py [hunk 1]
```diff
@@ -799,12 +799,12 @@ async def send_transaction(self, request) -> EndpointResult:
         if "memos" in request:
             memos = [mem.encode("utf-8") for mem in request["memos"]]
 
-        if "fee" in request:
-            fee = uint64(request["fee"])
-        else:
-            fee = uint64(0)
+        fee: uint64 = uint64(request.get("fee", 0))
+        min_coin_amount: uint128 = uint128(request.get("min_coin_amount", 0))
```
Comment (mariano54) [inline chia/rpc/wallet_rpc_api.py:803]: coins are uint64, but total balance can be uint128

**Review by mariano54 — CHANGES_REQUESTED** [review]
Nice change! Does this work for NFT offers also?

**Review by mariano54 — APPROVED** [review]

**Review by trepca — APPROVED** [review]

**Pull request merged**